File: rust/plugins/system/android/src/main/java/app/nativeblade/system/NativeBladeSystemPlugin.kt
package app.nativeblade.system

import android.app.Activity
import android.content.Intent
import android.net.Uri
import android.provider.Settings
import app.tauri.annotation.Command
import app.tauri.annotation.TauriPlugin
import app.tauri.plugin.Invoke
import app.tauri.plugin.Plugin

@TauriPlugin
class NativeBladeSystemPlugin(private val activity: Activity) : Plugin(activity) {

    @Command
    fun openAppSettings(invoke: Invoke) {
        try {
            val intent = Intent(Settings.ACTION_APPLICATION_DETAILS_SETTINGS).apply {
                data = Uri.fromParts("package", activity.packageName, null)
                addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
            }
            activity.startActivity(intent)
            invoke.resolve()
        } catch (e: Exception) {
            invoke.reject(e.message ?: "failed to open app settings")
        }
    }
}

File: rust/plugins/system/ios/Sources/NativeBladeSystemPlugin.swift
import SwiftRs
import Tauri
import UIKit

class NativeBladeSystemPlugin: Plugin {
    @objc public func openAppSettings(_ invoke: Invoke) {
        DispatchQueue.main.async {
            guard let url = URL(string: UIApplication.openSettingsURLString) else {
                invoke.reject("failed to build the app settings URL")
                return
            }
            UIApplication.shared.open(url, options: [:]) { success in
                if success {
                    invoke.resolve()
                } else {
                    invoke.reject("failed to open the app settings URL")
                }
            }
        }
    }
}

@_cdecl("init_plugin_nativeblade_system")
func initPlugin() -> Plugin {
    return NativeBladeSystemPlugin()
}
